src/main/java/com/celements/model/access/DefaultModelAccessFacade.java

```diff
@@ -633,6 +634,38 @@ private BaseProperty getBaseProperty(BaseObject obj, String key) {
   }
 
   @Override
+  public <T> Optional<T> getFieldValue(XWikiDocument doc, ClassField<T> field) {
+    return Optional.fromNullable(getProperty(checkNotNull(doc), checkNotNull(field)));
+  }
+
+  @Override
+  public <T> Optional<T> getFieldValue(DocumentReference docRef, ClassField<T> field)
+      throws DocumentNotExistsException {
+    return Optional.fromNullable(getProperty(checkNotNull(docRef), checkNotNull(field)));
```

Comment:
This is right. Over four more embedded method calls in the current implementation. Yet, fast failure means checking early.